components/admin_console/session_length_settings.jsx

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import * as Utils from 'utils/utils.jsx';
+
+import AdminSettings from './admin_settings';
+import BooleanSetting from './boolean_setting';
+import SettingsGroup from './settings_group.jsx';
+import TextSetting from './text_setting';
+
+export default class SessionLengthSettings extends AdminSettings {
+    getConfigFromState = (config) => {
+        const MINIMUM_IDLE_TIMEOUT = 5;
+
+        config.ServiceSettings.ExtendSessionLengthWithActivity = this.state.extendSessionLengthWithActivity;
+        config.ServiceSettings.SessionLengthWebInDays = this.parseIntNonZero(this.state.sessionLengthWebInDays);
+        config.ServiceSettings.SessionLengthMobileInDays = this.parseIntNonZero(this.state.sessionLengthMobileInDays);
+        config.ServiceSettings.SessionLengthSSOInDays = this.parseIntNonZero(this.state.sessionLengthSSOInDays);
+        config.ServiceSettings.SessionCacheInMinutes = this.parseIntNonZero(this.state.sessionCacheInMinutes);
+        // eslint-disable-next-line no-magic-numbers
```

Comment:
Is this eslint comment still required?

Reply:
Checked the cypress folder quickly and didn't find any session expiration related test so probably no e2e tests so far. However, it would be great if they could add it with this PR.

Reply:
@mkraft Nope, I don't think so.